src/Event/EuLoginEventSubscriber.php

```diff
@@ -67,12 +67,33 @@ public static function create(ContainerInterface $container) {
    */
   public static function getSubscribedEvents(): array {
     $events = [];
-    $events[CasHelper::EVENT_PRE_REGISTER] = 'processUserProperties';
+    $events[CasHelper::EVENT_PRE_REGISTER] = [
+      ['checkUserMailExists', 1000],
+      ['processUserProperties'],
+    ];
     $events[CasHelper::EVENT_POST_VALIDATE] = 'processCasAttributes';
     $events[CasHelper::EVENT_PRE_VALIDATE] = 'alterValidationPath';
     return $events;
   }
 
+  /**
+   * Checks user e-mail exists previously.
```

Comment:
s/'Checks'/'Checks that'. Also there is a Drupal convention to use 'email' rather than 'e-mail'. See https://www.drupal.org/drupalorg/style-guide/content#relatedwords